### PR TITLE
Fix detection of invalid context files when importing

### PR DESCRIPTION
**Pull request by thaJeztah**
fixes https://github.com/docker/cli/issues/1920

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @goksu @simonferquel @silvin-lubecki PTAL

**Comment by silvin-lubecki** [comment]
Linter is complaining 😅 
```
cli/context/store/store_test.go:172:10:warning: ineffectual assignment to err (ineffassign)
```

File: cli/context/store/store_test.go [hunk 1]
```diff
@@ -144,6 +145,38 @@ func TestDetectImportContentType(t *testing.T) {
...
+		Size: int64(len("hello world")),
+	}
+	err = tw.WriteHeader(hdr)
+	assert.NilError(t, err)
+	_, err = tw.Write([]byte("hello world"))
+	assert.NilError(t, err)
+	err = tw.Close()
+	assert.NilError(t, err)
+
+	source, err := os.Open(tf)
```
Comment (thaJeztah) [inline cli/context/store/store_test.go]: Adding an assert here to fix the linting failure

```suggestion
	source, err := os.Open(tf)
	assert.NilError(t, err)
```

**Comment by thaJeztah** [comment]
opened a backport for when this is merged https://github.com/docker/cli/pull/1929

**Pull request merged**